extension/background/intentRunner.js

```diff
@@ -525,32 +525,27 @@ registerHandler("getRegisteredNicknames", getRegisteredNicknames);
 
 let pageNames = {};
 
+export async function getRegisteredPageName() {
+  const result = await browser.storage.sync.get("pageNames");
+  pageNames = result.pageNames;
+
+  return pageNames;
+}
+
 export async function registerPageName(name, { url }) {
```

Comment:
I assume this is the function where I am to save the pageNames to the entityTypes, it's not clear to me how the services communicate

Reply:
Yes, this should call a new function in `entityTypes.js`